views/news/news-container.js

```diff
@@ -21,6 +22,98 @@ import * as c from '../components/colors'
 let Entities = require('html-entities').AllHtmlEntities
 const entities = new Entities()
 
+type NewsRowPropsType = {
+  story: StoryType,
+  onPress: (title: string, story: StoryType) => any,
+};
+
+class NewsRow extends React.Component {
+  state = {
+    thumbnailUrl: null,
+  }
+
+  componentWillMount() {
+    this.makeThumbnail(this.props)
+  }
+
+  componentWillReceiveProps(nextProps: NewsRowPropsType) {
+    this.makeThumbnail(nextProps)
+  }
+
+  props: NewsRowPropsType
+
+  makeThumbnail = async (props: NewsRowPropsType) => {
+    const thumbnailUrl = await new Promise(resolve => {
+      // grab the URL
+      const url = this.findImage(props.story.content)
+      // if we didn't find a valid URL, return null
+      if (!url) {
+        return null
```

Comment:
What about a `return` after the `resolve(null`? It seems to behave well.

      if (!url) {
        resolve(null)
        return
      }